app/controllers/Leagues.java

```diff
@@ -13,214 +11,213 @@
+		SearchFormData st = new SearchFormData();
+		Form<SearchFormData> stuff = Form.form(SearchFormData.class).fill(st);
+
+		Form<SearchFormData> searcher = Form.form(SearchFormData.class).bindFromRequest();
+		SearchFormData st2 = searcher.get();
+
+		Page<League> currPage = League.find(st2.term, "leagueName asc", page);
+
+		return ok(SearchLeagues.render("Leagues", currPage, Secured.isLoggedIn(ctx()), st2.term));
+	}
+
+	public static Result editSchedule(){
+		LeagueForm leagueForm = new LeagueForm();
+		Form<LeagueForm> emptyForm = Form.form(LeagueForm.class).fill(leagueForm);
+		return ok(LeagueSchedule.render("Create Schedule", emptyForm, Secured.isLoggedIn(ctx())));
+	}
+
+	public static void generateSchedule(League league){
+
+		List<Team> opp = league.getTeams();
+		List<Team> home = new ArrayList<Team>();
+		List<Team> away = new ArrayList<Team>();
+
+		if(opp == null || opp.size() == 1){
+
+		}
+		else if(opp.size() < 3){
+			for(int i = 0; i < league.getDateList().size(); i++){
+				List<String> dates = league.getDateList();
+				league.addOpponent(dates.get(i), opp.get(0).getTeamName(), opp.get(1).getTeamName());
+			}
+		}
+		else{
+			for(int i = 0, j = opp.size() / 2; i < opp.size() / 2; i++, j++){
+				home.add(opp.get(i));
+				away.add(opp.get(j));
+				if((i+1)*2+1 == opp.size()){
+					home.add(opp.get((i+1)*2));
+					away.add(null);
+				}
+			}
+
+
+			List<String> dates = league.getDateList();
+			for(int i = 0; i < league.getDateList().size(); i++){
+				for(int j = 0; j < home.size(); j++){
+					if(away.get(j) != null && home.get(j) != null){
+						league.addOpponent(dates.get(i), home.get(j).getTeamName(), away.get(j).getTeamName());
+					}
+					else if(away.get(j) == null){
+						league.addOpponent(dates.get(i), home.get(j).getTeamName(), "null");
+					}
+					else {
+						league.addOpponent(dates.get(i), away.get(j).getTeamName(), "null");
+					}
+				}
+				away.add(home.remove(home.size() -1));
+				home.add(1, away.remove(0));
+			}
+		}
+
+	}
+
+	public static Result editLeague(long id){
+		League league = LeagueDB.getLeague(id);
+		Form<LeagueForm> tempForm = Form.form(LeagueForm.class).bindFromRequest();
+		int regStep = league.getRegStep();
+
+		switch(regStep){
+		case 1:
+			league.setNumTeams(tempForm.get().numTeams);
+			league.setNumGames(tempForm.get().numGames);
+			league.setRegStep(2);
+			break;
+		case 2:
+			league.setDateList(tempForm.get().dateList);
+			league.setRegStep(3);
+			break;
+		case 3:
+			List<String> teamList = tempForm.get().teams;
+
+			for(int i = 0; i < teamList.size(); i++){
+				if(!league.getTeams().contains(Team.getTeam(teamList.get(i)))){
+					league.addTeam(Team.getTeam(teamList.get(i)));
+				}
+			}
+			league.eraseSchedule();
+			generateSchedule(league);
+			league.setRegStep(4);
+			break;
+		case 4:
+			league.setCourt(Court.getCourt(tempForm.get().court));
+			league.setDescription(tempForm.get().description);
+			league.setRegStep(5);
+			break;
+		case 5:
```

Comment:
Yeah good call. Looking at my old code is painful man.  Everything is so hacky lol.  I know I rushed through a lot of this too.